src/app/services/context/popups-context/PopupsContext.tsx

```diff
@@ -0,0 +1,78 @@
+import { createContext, useContext, useReducer } from 'react';
+
+import { PopupsComponent, PopupsProps, PopupsStatus, PopupsType } from '.';
+
+export interface Popup<Data = {}> {
+  component: React.ElementType<PopupsComponent<Data>>,
+  popupsProps: PopupsProps<Data>,
+  type: PopupsType,
+  id: string;
+  result?: any;
+  open?: boolean;
```

Comment:
What's the difference between open and status?

Reply:
Open can be true / false
Status can be used to tell the caller what the popup is doing. 
Has it been closed? (should be used when an action in the popup was successfully completed)
Has it been dismissed? (should be used when no action was done / action was unsuccessful)